### PR TITLE
feat(compat): add Vector3 to-ref helpers

### PR DESCRIPTION
**Pull request by ikkou**
## Summary

- Add Babylon.js-style `Vector3` `*ToRef` helpers to `@babylonjs/lite-compat`
- Route the existing allocating helpers through the new `ToRef` implementations
- Cover the helpers with compat unit tests, including zero-vector normalization and result reuse

## Why

Babylon.js exposes `ToRef` variants for common `Vector3` operations so callers can reuse an existing result vector instead of allocating a new one. The compat layer already had the allocating forms (`Cross`, `Lerp`, `Normalize`, `TransformCoordinates`, etc.) and `CenterToRef`, but several matching `ToRef` helpers were missing.

Adding these small helpers improves source compatibility for Babylon.js code that already uses `Vector3.*ToRef(...)` while keeping the implementation local to the math compat surface.

## How I noticed

After looking for a non-doc follow-up contribution, I checked the compat math surface because `COMPAT-STATUS.md` marks `Vector2` / `Vector3` / `Vector4` as full and already calls out some `Vector3` helpers. Comparing the existing compat `Vector3` implementation with Babylon.js' common `ToRef` pattern showed this small gap.

## Validation

- `corepack pnpm vitest run --project compat packages/babylon-lite-compat/tests/math.test.ts`
- `corepack pnpm vitest run --project compat`
- `git diff --check`
- `corepack pnpm lint`
- `corepack pnpm --filter @babylonjs/lite-compat build`

### CODE REVIEW

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260618.60/lab/index.html)

<sub>Build 20260618.60 - merge @ 8737051b1692dc88e693a5a13ae6a2f9d90019ba</sub>

**Comment by bjsplat** [comment]
## Bundle Size Changes

### Decreases

| Package | Current | Master | Change |
|---------|---------|--------|--------|
| Scene 9 — Sponza (.babylon)<br/>`scene9` | 58 KB | 59 KB | -1 KB |
| Scene 17 — PBR + Std TI<br/>`scene17` | 84 KB | 85 KB | -1 KB |
| Scene 45 — Physics Collision Filtering<br/>`scene45` | 50 KB | 51 KB | -1 KB |
| Scene 46 — Physics Constraints<br/>`scene46` | 53 KB | 54 KB | -1 KB |
| Scene 48 — Physics Center of Mass<br/>`scene48` | 53 KB | 54 KB | -1 KB |
| Scene 49 — Physics Shape Queries<br/>`scene49` | 91 KB | 92 KB | -1 KB |
| Scene 53 — Depth-Hosted Sprites Mixed With 3D<br/>`scene53` | 57 KB | 58 KB | -1 KB |
| Scene 154 - Step Time Animation<br/>`scene154` | 54 KB | 55 KB | -1 KB |
| Scene 171 - Navigation Crowd Path<br/>`scene171` | 94 KB | 95 KB | -1 KB |
| Scene 175 - Navigation Raycast<br/>`scene175` | 93 KB | 94 KB | -1 KB |
| Scene 202 - LWR Point Light<br/>`scene202` | 48 KB | 49 KB | -1 KB |
| Scene 209 — LWR Physics<br/>`scene209` | 53 KB | 54 KB | -1 KB |
| Scene 227 — Multi-Canvas (Same Scene)<br/>`scene227` | 47 KB | 48 KB | -1 KB |

*Sizes rounded to nearest KB. Run `pnpm build:bundle-scenes` locally to verify.*

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260619.1/lab/index.html)

<sub>Build 20260619.1 - merge @ 8737051b1692dc88e693a5a13ae6a2f9d90019ba</sub>

File: packages/babylon-lite-compat/src/math/vector.ts [hunk 1]
```diff
@@ -291,19 +311,44 @@ export class Vector3 {
...
     }
 
     public static Maximize(a: Vector3, b: Vector3): Vector3 {
-        return new Vector3(Math.max(a.x, b.x), Math.max(a.y, b.y), Math.max(a.z, b.z));
+        return Vector3.MaximizeToRef(a, b, new Vector3());
+    }
+
+    public static MaximizeToRef<T extends Vector3>(a: Vector3, b: Vector3, result: T): T {
+        result.set(Math.max(a.x, b.x), Math.max(a.y, b.y), Math.max(a.z, b.z));
+        return result;
```
Comment (ryantrem) [inline packages/babylon-lite-compat/src/math/vector.ts]: This looks like there was an error in the original version (even before adding the ToRef variants). It seems like what was meant to be here is `minimizeInPlace`, `minimizeInPlaceFromFloats`, `maximizeInPlace`, `maximizeInPlaceFromFloats`?

**Review by ryantrem — APPROVED** [review]

**Comment by bjsplat** [comment]
## Bundle Size Changes

### Increases

| Package | Current | Master | Change |
|---------|---------|--------|--------|
| Scene 215 — Cascaded Shadow Maps (PBR)<br/>`scene215` | 78 KB | 76 KB | **+2 KB** |
| Scene 14 — Flight Helmet<br/>`scene14` | 89 KB | 88 KB | **+1 KB** |
| Scene 33 — KHR_lights_punctual<br/>`scene33` | 99 KB | 98 KB | **+1 KB** |
| Scene 47 — Physics Heightfield<br/>`scene47` | 89 KB | 88 KB | **+1 KB** |
| Scene 73 — Wheel PBR vs NME Viewports<br/>`scene73` | 142 KB | 141 KB | **+1 KB** |
| Scene 104 — Physics Character Controller<br/>`scene104` | 100 KB | 99 KB | **+1 KB** |
| Scene 111 — Light Selection Stress Test<br/>`scene111` | 134 KB | 133 KB | **+1 KB** |
| Scene 147 — Circle of Confusion<br/>`scene147` | 101 KB | 100 KB | **+1 KB** |
| Scene 211 — BrainStem Meshopt<br/>`scene211` | 86 KB | 85 KB | **+1 KB** |
| Scene 214 — Cascaded Shadow Maps<br/>`scene214` | 66 KB | 65 KB | **+1 KB** |

*Sizes rounded to nearest KB. Run `pnpm build:bundle-scenes` locally to verify.*

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260620.1/lab/index.html)

<sub>Build 20260620.1 - merge @ 93df7d04049f93f181ccfbf42e69c0906a613595</sub>
